### PR TITLE
server: implement CHAOS-only dns Exchanger

### PR DESCRIPTION
**Pull request by amery**

### CODE REVIEW

File: pkg/server/server_resolver.go [hunk 1]
```diff
@@ -33,5 +37,24 @@ func (*Server) reflectEnabled(_ string) (slog.LogLevel, bool) {
 
 // ServeDNS handles dns requests based on the IP address of the client
 func (srv *Server) ServeDNS(rw dns.ResponseWriter, req *dns.Msg) {
+	var chaos bool
```
Comment (karasz) [inline pkg/server/server_resolver.go:40]: It should be possible to explicitly disable the `CHAOS` protocol as some users might not want to reveal the server type

**Review by karasz — CHANGES_REQUESTED** [review]

File: pkg/server/server_resolver.go (hunk 1 above)
Comment (amery) [reply pkg/server/server_resolver.go:40]: What should we return if CHAOS is disabled? Not implemented or an empty Success?

**Comment by amery** [comment]
added `DisableCHAOS` flag making the DNS server return an empty `dns.RcodeSuccess` instead of answering Authors/Version/Hostname.

**Review by karasz — APPROVED** [review]
LGTM

**Pull request merged**